hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -206,9 +206,12 @@ public void run() {
 
     @Override
     public void clearAllListeners() {
-        for (Runnable removeAction : removeListenerActions) {
+        //Changed from normal iteration to copying with toArray because of ConcurrentModificationException.
+        // toArray is called under internal mutex of synchronized list.
+        Object[] actions = removeListenerActions.toArray();
+        for (Object removeAction : actions) {
             try {
-                removeAction.run();
+                ((Runnable) removeAction).run();
             } catch (Exception e) {
                 getLogger().warning("Exception during destroy action", e);
```

Comment:
message is misleading, how about `Exception while running remove-listener-action` or something better ?